portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -36,25 +37,48 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private static final int TIME_DESCENDING = 0;
+  private static final int TIME_ASCENDING = 1;
+
+  private int numDisplay = 10;  // Default value.
+  private int sortOrder;
+
   /**
    * Handles a GET request, and fetches all the comments data from the Datastore.
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    boolean toChange = getToChange(request);
+    if (toChange) {
+      numDisplay = Integer.parseInt(request.getHeader("numComments"));
+      sortOrder = getSortOrder(request);
+    }
+
     Query query = new Query("Comment").addSort("time", SortDirection.DESCENDING);
+    if (sortOrder == TIME_ASCENDING) {
+      query = new Query("Comment").addSort("time", SortDirection.ASCENDING);
+    }
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
     List<String> comments = new ArrayList<>();
+    int index = 0;
     for (Entity entity : results.asIterable()) {
       String name = (String) (entity.getProperty("name"));
       String comment = (String) (entity.getProperty("comment"));
 
       final String result = name + ": " + comment;
       comments.add(result);
+      
+      index++;
+      if (index >= numDisplay) {
+        break;
+      }
     }
```

Comment:
[Optional] I had a similar discussion w/ Nathan but I prefer the style where you use a for loop over the index variable and break if the results iterable is empty.  This avoid the ugly index++ line and instead uses results.next() which I find more readable.  That said, after seeing the end results in Nathan's code I don't think there is too much of a difference so you can leave it like this if you want.